src/core/types/Service.ts

```diff
@@ -55,6 +56,9 @@ export interface VaultService {
   deposit: (props: DepositProps) => Promise<TransactionResponse>;
   withdraw: (props: WithdrawProps) => Promise<TransactionResponse>;
   migrate: (props: MigrateProps) => Promise<TransactionResponse>;
+  approveDeposit: (props: ApproveDepositProps) => Promise<TransactionResponse | Boolean>;
+  approveZapOut: (props: ApproveZapOutProps) => Promise<TransactionResponse | Boolean>;
```

Comment:
With new refactor on sdk, we can remove the `Boolean` from return type
```suggestion
  approveDeposit: (props: ApproveDepositProps) => Promise<TransactionResponse>;
  approveZapOut: (props: ApproveZapOutProps) => Promise<TransactionResponse>;
```

Reply:
Also, we should always use lowercase `boolean` as that's the primitive type (not `Boolean`).

Reply:
Yes, that was an error on the SDK that was removed in the recent version, so Fibo probably was forced to copy that interface. Agree we should stick with primitive. Also we should not have this type of interfaces that can return multiple types.

Reply:
fixed on 227dce8ccc7427637245d1ffbe9f21f0819cece8